sdcflows/interfaces/bspline.py

```diff
@@ -123,16 +127,45 @@ class BSplineApprox(SimpleInterface):
 
     def _run_interface(self, runtime):
         from sklearn import linear_model as lm
-        from scipy.sparse import vstack as sparse_vstack
+
+        # Output name baseline
+        out_name = fname_presuffix(
+            self.inputs.in_data, suffix="_field", newpath=runtime.cwd
+        )
 
         # Load in the fieldmap
         fmapnii = nb.load(self.inputs.in_data)
+        zooms = fmapnii.header.get_zooms()
+
+        # Get a mask (or define on the spot to cover the full extent)
+        masknii = (
+            nb.load(self.inputs.in_mask)
+            if isdefined(self.inputs.in_mask)
+            else None
+        )
+
+        need_resize = np.any(np.array(zooms) < self.inputs.zooms_min)
+        if need_resize:
+            from niworkflows.utils.images import resample_by_spacing
+
+            LOGGER.info(
+                "Resampling image with resolution exceeding 'zooms_min' "
+                f"({'x'.join(str(s) for s in zooms)})."
+            )
+            fmapnii = resample_by_spacing(fmapnii, [self.inputs.zooms_min] * 3)
```

Comment:
I have [general problems](https://github.com/nipreps/niworkflows/issues/696) with resample_by_spacing. Here I think it is massive overkill. If we can agree on a safe factor-of-2 range of okay knot-to-voxel ratios, then there's a much simpler approach, where we low-pass filter and downsample without interpolation.

I found the function I was using during testing:

```Python
def _blur(in_file, target_zooms):
    import os
    import numpy as np
    import nibabel as nb
    from scipy import ndimage as ndi
    from nipype.utils.filemanip import fname_presuffix

    img = nb.load(in_file)
    factors = target_zooms[:3] / np.array(img.header.get_zooms()[:3])

    # Downsample by 2 or not at all
    scale = np.clip(factors.astype(int) // 4, 1, 2)

    if not np.any(scale > 1):
        return in_file

    out_filename = fname_presuffix(in_file, suffix='blur', newpath=os.getcwd())

    blurred = img.__class__(
        ndi.gaussian_filter(img.get_fdata(), (scale - 1) / 2),
        affine=img.affine,
        header=img.header
    )

    slicers = tuple(
        slice(1, None, 2) if factor == 2 else slice(None) for factor in scale
    )

    out = blurred.slicer[slicers]
    out.to_filename(out_filename)
    return out_filename
```

Reply:
Note that your current approach will also resample 1.2x2.0x2.0 to 1.4x1.4x1.4. The above approach would resample to 2.4x2.0x2.0.

Reply:
- smoothing: I don't think that's necessary - the scipy interpolation function has a `filter` option that will smooth for you.
- the problem of resolution is indeed something I'll fix. Before you even pointed at the problem, I've been talking with @eilidhmacnicol that you definitely want to be able to keep anisotropic voxels for rats. So I guess it is more about accepting either one or three floats as input.

Reply:
+1 on avoiding using the current implementation of `resample_by_spacing`

Reply:
I have implemented an alternative function. As I said before, I would leave more specialized blurring and zooms ratios for a previous interface if necessary.